### PR TITLE
Store currently selected tab in package state

### PR DESCRIPTION
**Pull request by josa42**
Also fixes: #660

### CODE REVIEW

**Comment by steelbrain** [comment]
@josa42 does it work? 'cause the code looks good to me.

**Comment by josa42** [comment]
@steelbrain Yea, I'm done

**Comment by steelbrain** [comment]
How about you apply this patch to your branch
```diff
diff --git a/lib/linter-views.coffee b/lib/linter-views.coffee
index f707b96..6c7786b 100644
--- a/lib/linter-views.coffee
+++ b/lib/linter-views.coffee
@@ -108,7 +108,7 @@ class LinterViews
     unless @tabs[@state.scope]?.visibility
       @state.scope = @tabPriority.filter((key) => @tabs[key].visibility)[0]
 
-    @changeTab(@state.scope)
+    @changeTab(@state.scope, false)
 
   # consumed in editor-linter, renderPanel
   updateBubble: (point) ->
@@ -159,7 +159,7 @@ class LinterViews
       item: @bottomStatus,
       priority: -999
 
-  changeTab: (tabName) ->
+  changeTab: (tabName, render = true) ->
     @showPanel = not @tabs[tabName]?.active
     if not @showPanel
       for key, tab of @tabs
@@ -168,7 +168,7 @@ class LinterViews
       @state.scope = tabName
       for key, tab of @tabs
         tab.active = tabName is key
-      @renderPanelMessages()
+      @renderPanelMessages() if render
     @setShowPanel @showPanel
 
   removeBubble: ->
```

**Comment by josa42** [comment]
@steelbrain :+1:

File: lib/linter-views.coffee [hunk 1]
```diff
@@ -94,22 +93,23 @@ class LinterViews
...
-    @changeTab(firstLabel) if first and not foundActive
+        first = tab unless first
+
     first.classList.add('first') if first
     last.classList.add('last') if last
 
+    unless @tabs[@state.scope]?.visibility
+      @state.scope = @tabPriority.filter((key) => @tabs[key].visibility)[0]
+
+    @changeTab(@state.scope, false)
```
Comment (steelbrain) [inline lib/linter-views.coffee]: How does it handle if there's no active tab btw?
Does it still show the results from the last active tab?

Comment (josa42) [inline lib/linter-views.coffee]: If no tab is activated messages from the current file are shown.

Comment (josa42) [inline lib/linter-views.coffee]: But the @state.scope should probably be 'File' not undefined.

**Comment by steelbrain** [comment]
LGTM

**Pull request merged**